analyzer/runtime/runtime.go

```diff
@@ -228,47 +227,41 @@ func (m *Main) prework() error {
 func (m *Main) processRound(ctx context.Context, round uint64) error {
 	ctxWithTimeout, cancel := context.WithTimeout(ctx, ProcessRoundTimeout)
 	defer cancel()
-	group, groupCtx := errgroup.WithContext(ctxWithTimeout)
+
+	// Fetch all data.
+	data, err := m.cfg.Source.AllData(ctxWithTimeout, round)
+	if err != nil {
+		if strings.Contains(err.Error(), "roothash: block not found") {
+			return analyzer.ErrOutOfRange
+		}
+		return err
+	}
 
 	// Prepare and perform updates.
 	batch := &storage.QueryBatch{}
-
-	type prepareFunc = func(context.Context, uint64, *storage.QueryBatch) error
+	type prepareFunc = func(*storage.QueryBatch, *storage.RuntimeAllData) error
+	// Process block and event data.
 	for _, f := range []prepareFunc{
```

Comment:
Optional: Now that we don't have goroutines any more, maybe just explicitly call the two funcs and skip the for loop? You call on what's more readable.

Reply:
Hmm I think I slightly prefer the loop to avoid repeating the error handling boilerplate. Also I'm hoping that we'll eventually re-add goroutines (gated by some flag) for faster indexing performance once the indexer is more stable.